src/spikeinterface/preprocessing/zero_channel_pad.py

```diff
@@ -1,10 +1,115 @@
+
+    Parameters
+    ----------
+    parent_recording_segment : BaseRecording
+        The parent recording segment from which the traces are to be retrieved.
+    padding_start : int
+        The amount of padding to add to the left of the traces. Default is 0. It has to be non-negative
+    padding_end : int
+        The amount of padding to add to the right of the traces. Default is 0. It has to be non-negative
+    fill_value: float
+        The value to pad with. Default is 0.
+    """
+
+    def __init__(
+        self, parent_recording: BaseRecording, padding_start: int = 0, padding_end: int = 0, fill_value: float = 0.0
+    ):
+        assert padding_end >= 0 and padding_start >= 0, "Paddings must be >= 0"
+        super().__init__(recording=parent_recording)
+
+        self.padding_start = padding_start
+        self.padding_end = padding_end
+        self.fill_value = fill_value
+        for segment in parent_recording._recording_segments:
+            recording_segment = TracePaddedRecordingSegment(
+                segment,
+                parent_recording.get_num_channels(),
+                self.dtype,
+                self.padding_start,
+                self.padding_end,
+                self.fill_value,
+            )
+            self.add_recording_segment(recording_segment)
+
+        self._kwargs = dict(
+            parent_recording=parent_recording,
+            padding_start=padding_start,
+            padding_end=padding_end,
+            fill_value=fill_value,
+        )
+
+
+class TracePaddedRecordingSegment(BasePreprocessorSegment):
+    def __init__(
+        self,
+        parent_recording_segment: BaseRecordingSegment,
+        num_channels,
+        dtype,
+        paddign_left,
+        padding_end,
+        fill_value,
+    ):
+        self.padding_start = paddign_left
+        self.padding_end = padding_end
+        self.fill_value = fill_value
+        self.num_channels = num_channels
+        self.num_samples_in_original_segment = parent_recording_segment.get_num_samples()
+        self.dtype = dtype
+
+        super().__init__(parent_recording_segment=parent_recording_segment)
+
+    def get_traces(self, start_frame, end_frame, channel_indices):
+        if start_frame is None:
+            start_frame = 0
+        if end_frame is None:
+            end_frame = self.get_num_samples()
+
+        # This contains the padded elements by default and we add the original traces if necessary
+        trace_size = end_frame - start_frame
+        if isinstance(channel_indices, (np.ndarray, list)):
+            num_channels = len(channel_indices)
+        elif channel_indices == slice(None):
+            num_channels = self.num_channels
+        else:
+            raise ValueError(f"Unsupported channel_indices type: {type(channel_indices)} raise an issue on github ")
+
+        output_traces = np.full(shape=(trace_size, num_channels), fill_value=self.fill_value, dtype=self.dtype)
```

Comment:
I think that if we are not one the border we could avoid this buffer alocation by returning directly the original_traces shifted.
This buffer only make sens on the left/right border.

Reply:
I thought about that.

Honestly, I find it annoying that I have to reduce sequential readiability for the sake of memory optimizations out here but it probably makes sense. I wish that `get_traces` would have an out method like ufunctions in numpy so we could just pass previously allocated memory.

Reply:
Check this out:

#1735

Reply:
That will a better solution but sadly I don't have time. I implemented you suggestion, check out the last PR.